validation/analyze_qam.py

```diff
@@ -0,0 +1,152 @@
+    parser.add_argument('-c', '--cores', default=2, type=int,
+                        help='Number of cores for parallel computation')
+    return parser.parse_args()
+
+def generate_latex_table(data_esa, data_nersc, out_path, data_name, print_RQM = True):
+    """
+    :param data_esa: json file with RQM for ESA data
+    :param data_nersc: json file with RQM for NERSC data
+    :param print_RQM: print RQM in the LATEX table
+    :param out_path: output folder
+    :param data_name: base name for output files
+    :param print_RQM statisctics in Latex table (additional table will be created)
+
+    :return: LATEX table in txt format
+             PNG file with error bars
+    """
+
+    with open(data_esa) as esa_json_file:
+        d_esa = json.load(esa_json_file)
+
+    with open(data_nersc) as nersc_json_file:
+        d_nersc = json.load(nersc_json_file)
+
+    print('\n%s %s\n' % (d_esa, d_nersc))
+
+    rows = []
+    rows.append(['No.', 'Scene ID'])
+
+    if print_RQM:
+        rows_RQM = []
+        rows_RQM.append(['No.', 'Scene ID', 'ESA', 'NERSC'])
+        total_mean_esa = []
+        total_mean_nersc = []
+    else:
+        rows.append(['No.', 'Scene ID'])
+
+    for i, sar_id in enumerate(d_esa):
+        mean_esa = d_esa[sar_id]['Mean'][0]
+        mean_nersc = d_nersc[sar_id]['Mean'][0]
+        total_mean_esa.append(mean_esa)
+        total_mean_nersc.append(mean_nersc)
+
+        if print_RQM:
+            rows_RQM.append([i, sar_id, '%.3f' % mean_esa,
+                         '%.3f' % mean_nersc])
+
+        rows.append([i, sar_id])
+
+    if print_RQM:
+        rows_RQM.append(['Total', '', '%.3f' % np.nanmean(total_mean_esa),
+                     '%.3f' % np.nanmean(total_mean_nersc)])
+        rows_RQM.append(['STD', '', '%.3f' % np.nanstd(total_mean_esa),
+                     '%.3f' % np.nanstd(total_mean_nersc)])
+
+    if print_RQM:
+        table_RQM = Texttable()
+        table_RQM.set_cols_align(["c"] * 4)
+        table_RQM.set_deco(Texttable.HEADER | Texttable.VLINES)
+        table_RQM.add_rows(rows_RQM)
+
+    table = Texttable()
+    table.set_cols_align(["c"] * 2)
+    table.set_deco(Texttable.HEADER | Texttable.VLINES)
+    table.add_rows(rows)
+
+    # Save table to file
+    if print_RQM:
+        with open('%s/%s_table_RQM.txt' % (out_path, data_name), 'w') as f:
+            f.write(tabulate(rows_RQM, headers='firstrow', tablefmt='latex'))
+
+    with open('%s/%s_table.txt' % (out_path, data_name), 'w') as f:
+        f.write(tabulate(rows, headers='firstrow', tablefmt='latex'))
+
+    # Calculate mean difference
+    mean_diff = np.array(total_mean_esa) - np.array(total_mean_nersc)
+
+    return np.nanmean(total_mean_esa), np.nanmean(total_mean_nersc),\
+           np.nanstd(total_mean_esa), np.nanstd(total_mean_nersc), np.nanmean(mean_diff), np.nanstd(mean_diff)
+
+def plot_results(mean_esa, mean_nersc, std_esa, std_nersc, mean_diff, mean_diff_std, out_path, data_name):
```

Comment:
It is better to generate one bar plot with all areas in Python, rather than combine these plots in Latex. In latext captions and labels are not visible if you reduce the size of individual plots to fit everything in one figure.